src/discord/interaction_handlers/application_commands.rs

```diff
@@ -563,6 +568,36 @@ async fn handle_update_user_info(
     }
 }
 
+async fn handle_user_profile(
```

Comment:
and this function is the meat of the MR, it's where the actual responding happens